src/tactic/linear_combination.lean

```diff
@@ -0,0 +1,386 @@
+when config.normalize config.normalization_tactic
+
+/-! ### Part 4: Completed Tactic -/
+
+
+/--
+This is a tactic that attempts to prove the target by creating and applying a
+  linear combination of a list of equalities.  (If the `normalize` field of the
+  configuration is set to ff, then the tactic will simply set the user up to
+  prove their target using the linear combination instead of attempting to
+  finish the proof.)
+
+Note: The left and right sides of all the equations should have the same
+  ring type, and the coefficients should also have this type.  This type must
+  have the has_mul and add_group properties.  Also note that the target must
+  involve at least one variable.
+
+* Input:
+  * `heqs` : a list of names, referring to equations in the local context
+  * `coeffs` : a list of coefficients to be multiplied with the corresponding
+    equations in the list of names
+  * `config` : a linear_combination_config, which determines the tactic used
+      for normalization; by default, this value is the standard configuration
+      for a linear_combination_config
+
+* Output: tactic unit
+-/
+meta def linear_combination (heqs : list name) (coeffs : list pexpr)
+  (config : linear_combination_config := {}) : tactic unit :=
+do
+  hsum ← make_sum_of_hyps heqs coeffs,
+  hsum_on_left ← move_to_left_side hsum,
+  move_target_to_left_side,
+  set_goal_to_hleft_eq_tleft hsum_on_left,
+  prove_equal_if_desired config
+
+
+section interactive_mode
+setup_tactic_parser
+
+/--
+A parser that matches a pair in parentheses, where the first item in the pair
+is an identifier and the second item in the pair is a pexpr.
+
+* Input: None
+
+* Output: a lean.parser (name × pexpr)
+-/
+meta def parse_name_pexpr_pair : lean.parser (name × pexpr) :=
+do
+  tk "(",
+  id ← ident,
+  tk ",",
+  coeff ← parser.pexpr 0,
+  tk ")",
+  pure (id, coeff)
+
+/--
+`linear_combination` attempts to prove the target by creating and applying a
+  linear combination of a list of equalities.  The tactic will create a linear
+  combination by adding the equalities together from left to right, so the order
+  of the input hypotheses does matter.  If the `normalize` field of the
+  configuration is set to ff, then the tactic will simply set the user up to
+  prove their target using the linear combination instead of attempting to
+  finish the proof.
+
+Note: The left and right sides of all the equations should have the same
+  type, and the coefficients should also have this type.  This type must
+  have the has_mul and add_group properties.  Also note that the target must
+  involve at least one variable.
+
+* Input:
+  * `heqs` : a list of identifiers, referring to equations in the local context
+  * `coeffs` : a list of coefficients to be multiplied with the corresponding
+      equations in the list of names
+
+  * `input` : a sequence of name and pexpr pairs, which represents the pairs
+      of hypotheses and their corresponding coefficients
```

Comment:
@agoldb10 to clarify, in case you haven't seen this yet: this doc string will become the entry for this tactic in the [tactic docs](https://leanprover-community.github.io/mathlib_docs/tactics.html). So unlike the other doc strings here, which are written for people looking at the implementation, this one is user-facing.

Reply:
That makes sense, thank you both for pointing this out.  I just removed `pexpr` from this comment.